src/compiler/checker.ts

```diff
@@ -18474,6 +18460,9 @@ namespace ts {
 
         function checkTypeReferenceNode(node: TypeReferenceNode | ExpressionWithTypeArguments) {
             checkGrammarTypeArguments(node, node.typeArguments);
+            if (node.kind === SyntaxKind.TypeReference && node.typeName.jsdocDot && !isInJavaScriptFile(node) && !findAncestor(node, n => n.kind === SyntaxKind.JSDocTypeExpression)) {

```

Comment:
Another `findAncestor` call here. We shouldn't be checking a type reference in a jsdoc comment if we're not in a JS file anyway, right?